pkg/testutils/testutils.go

```diff
@@ -0,0 +1,227 @@
+func NewTestMock(t *testing.T, localObjs []runtime.Object) *Mocks {
+	mockctrl := gomock.NewController(t)
+	s := scheme.Scheme
+	if err := configv1.AddToScheme(s); err != nil {
+		t.Fatalf("Couldn't add configv1 scheme: (%v)", err)
+	}
+	if err := machineapi.AddToScheme(s); err != nil {
+		t.Fatalf("Couldn't add machine scheme: (%v)", err)
+	}
+	if err := cloudingressv1alpha1.SchemeBuilder.AddToScheme(s); err != nil {
+		t.Fatalf("Couldn't add cloudingressv1alpha1 scheme: (%v)", err)
+	}
+
+	ret := &Mocks{
+		FakeKubeClient: fake.NewFakeClientWithScheme(s, localObjs...),
+		MockCtrl:       mockctrl,
+		MockAws:        awsclient.NewMockClient(mockctrl),
+		Scheme:         s,
+	}
+
+	return ret
+}
+
+// CreateAPIServerObject creates a configv1.APIServer object
+func CreateAPIServerObject(clustername, clusterdomain string) *configv1.APIServer {
+	ret := &configv1.APIServer{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Spec: configv1.APIServerSpec{
+			ClientCA: configv1.ConfigMapNameReference{
+				Name: "",
+			},
+			ServingCerts: configv1.APIServerServingCerts{
+				NamedCertificates: []configv1.APIServerNamedServingCert{
+					{
+						Names: []string{fmt.Sprintf("api.%s", clusterdomain)},
+						ServingCertificate: configv1.SecretNameReference{
+							Name: fmt.Sprintf("%s-primary-certbundle-secret", clustername),
+						},
+					},
+				},
+			},
+		},
+	}
+	return ret
+}
+
+// CreateMachineObjectList makes a MachineList from the slice of names, and returns also a slice of Machine objects for convenience
+func CreateMachineObjectList(name []string, clusterid, role, region, zone string) (*machineapi.MachineList, []machineapi.Machine) {
+	machines := make([]machineapi.Machine, 0)
+	for _, n := range name {
+		machines = append(machines, CreateMachineObj(n, clusterid, role, region, zone))
+	}
+	ret := &machineapi.MachineList{
+		Items: machines,
+	}
+	return ret, machines
+}
+
+// CreateMachineObj makes a single AWS-style machinev1beta1.Machine object
+func CreateMachineObj(name, clusterid, role, region, zone string) machineapi.Machine {
+	ami := "ami-123456"
+	provider := &awsprovider.AWSMachineProviderConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "awsproviderconfig.openshift.io/v1beta1",
+			Kind:       "AWSMachineProviderConfig",
+		},
+		InstanceType:       "small",
+		BlockDevices:       []awsprovider.BlockDeviceMappingSpec{},
+		AMI:                awsprovider.AWSResourceReference{ID: &ami},
+		Tags:               []awsprovider.TagSpecification{{Name: fmt.Sprintf("kubernetes.io/cluster/%s", clusterid), Value: "owned"}},
+		IAMInstanceProfile: &awsprovider.AWSResourceReference{ID: pointer.StringPtr(fmt.Sprintf("%s-%s-profile", clusterid, role))},
+		UserDataSecret:     &corev1.LocalObjectReference{Name: "aws-cloud-credentials"},
+		Placement:          awsprovider.Placement{Region: region, AvailabilityZone: zone},
+		LoadBalancers: []awsprovider.LoadBalancerReference{
```

Comment:
That is weird, but a your change is reverted with `gofmt`. I'm not quite sure what's going on in the `gofmt` internals 🤷‍♀️ , so I'll reject this suggestion since I don't want to fight `gofmt`.

Reply:
Odd indeed.